atArgumentParser.go

```diff
@@ -2,102 +2,171 @@ package vmcommon
 
 import (
 	"encoding/hex"
-	"strings"
 )
 
-type atArgumentParser struct {
+// AtArgumentParser is a parser that splits arguments by @ character
+// [NotConcurrentSafe]
+type AtArgumentParser struct {
+	// First argument is a string (function name or hex-encoded bytecode), the rest are raw bytes
 	arguments [][]byte
-	code      []byte
 }
 
-const atSep = "@"
-const atSepChar = '@'
+const atSeparator = "@"
+const atSeparatorChar = '@'
+const startIndexOfConstructorArguments = 3
+const startIndexOfFunctionArguments = 1
+const minNumDeployArguments = 3
+const minNumCallArguments = 1
+const indexOfCode = 0
+const indexOfVMType = 1
+const indexOfCodeMetadata = 2
+const indexOfFunction = indexOfCode
```

Comment:
Should be set to `0` as well, even if it happens to be equal to `indexOfCode` at the moment. These two indices are used for different situations (deploy vs call) and should remain independent.

Reply:
Fixed.